File: cs/lc104.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int MaxDepth(TreeNode root) {
        if(root != null)
            return 1 + Math.Max(MaxDepth(root.left), MaxDepth(root.right));
        else
            return 0;
    }
    public int MaxDepthIterative(TreeNode root) {
        if(root == null)
            return 0;
        Queue<TreeNode> nodes = new Queue<TreeNode>();
        nodes.Enqueue(root);
        int depth = 0;
        while(nodes.Count > 0)
        {
            var nextLevel = new Queue<TreeNode>();
            while(nodes.Count > 0)
            {
                TreeNode n = nodes.Dequeue();
                if(n.left != null)
                    nextLevel.Enqueue(n.left);
                if(n.right != null)
                    nextLevel.Enqueue(n.right);
            }
            nodes = nextLevel;
            depth += 1;
        }
        return depth;
    }
}

File: python/lc104.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepthRecursion(self, root: Optional[TreeNode]) -> int:
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right)) if root else 0
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        nodes = [root]
        depth = 0
        while len(nodes) > 0:
            nextLevel = []
            while len(nodes) > 0:
                n = nodes.pop()
                if n.left:
                    nextLevel.append(n.left)
                if n.right:
                    nextLevel.append(n.right)
            nodes = nextLevel
            depth += 1
        return depth
